operators/src/util/raster_stream_to_geotiff.rs

```diff
@@ -0,0 +1,231 @@
+    let output_bounds = query_rect.bbox;
+
+    let driver = Driver::get("GTiff")?;
+    // TODO: "COMPRESS, DEFLATE" flags but rust-gdal doesn't support setting this yet(?)
+    let mut dataset =
+        driver.create_with_band_type::<T>(&file_name, width as isize, height as isize, 1)?;
+
+    dataset.set_spatial_ref(&spatial_reference.try_into()?)?;
+    dataset.set_geo_transform(&output_geo_transform.into())?;
+    let mut band = dataset.rasterband(1)?;
+
+    if let Some(no_data) = no_data_value {
+        band.set_no_data_value(no_data)?;
+    }
+
+    while let Ok(tile) = rx.recv() {
+        let tile_info = tile.tile_information();
+
+        let tile_bounds = tile_info.spatial_bounds();
+
+        let mat_tile = if output_bounds.contains_bbox(&tile_bounds) {
+            tile.into_materialized_tile()
+        } else {
+            // extract relevant data from tile (intersection with output_bounds)
+
+            // TODO: snap intersection to pixels?
+            let intersection = output_bounds
+                .intersection(&tile_bounds)
+                .expect("tile must intersect with query");
+
+            let shape: GridShape2D = [
+                (intersection.size_y() / y_pixel_size).ceil() as usize,
+                (intersection.size_x() / x_pixel_size).ceil() as usize,
+            ]
+            .into();
+
+            let output_grid = Grid2D::new_filled(
+                shape,
+                no_data_value.map_or_else(T::zero, T::from_),
+                no_data_value.map(T::from_),
+            );
+
+            let output_geo_transform = GeoTransform {
+                origin_coordinate: intersection.upper_left(),
+                x_pixel_size,
+                y_pixel_size: -y_pixel_size,
+            };
+
+            let mut output_tile = RasterTile2D::new_without_offset(
+                TimeInterval::default(),
+                output_geo_transform,
+                output_grid,
+            )
+            .into_materialized_tile();
+
+            output_tile.blit(tile)?;
+
+            output_tile
+        };
+
+        let upper_left = mat_tile.spatial_bounds().upper_left();
+
+        let upper_left_pixel_x = ((upper_left.x - output_geo_transform.origin_coordinate.x)
+            / x_pixel_size)
+            .floor() as isize;
+        let upper_left_pixel_y = ((output_geo_transform.origin_coordinate.y - upper_left.y)
+            / y_pixel_size)
+            .floor() as isize;
+        let window = (upper_left_pixel_x, upper_left_pixel_y);
+
+        let shape = mat_tile.grid_array.axis_size();
+        let window_size = (shape[1], shape[0]);
+
+        let buffer = Buffer::new(window_size, mat_tile.grid_array.data);
+
+        band.write(window, window_size, &buffer)?;
+    }
+
+    Ok(())
+}
+
+/// copies the bytes of the vsi in-memory file with given `file_name` and frees the memory
+fn get_vsi_mem_file_bytes_and_free(file_name: &str) -> Vec<u8> {
```

Comment:
please add a PR for rust-gdal :)

Reply:
I added a task https://trello.com/c/K8GSCmix . We shouldn't wait for a new gdal release to merge this PR

Reply:
ok